src/main/java/jarvis/commands/UserList.java

```diff
@@ -0,0 +1,146 @@
+        }
+    }
+
+    public static String getStatusIcon(int index){
+        return userList.get(index).getStatusIcon();
+    }
+
+    public static String getTask(int index) {
+        return userList.get(index).getDescription();
+    }
+
+
+    public static void markTask(int index, boolean toPrintMessage) {
+        Task targetTask = userList.get(index);
+        DisplayMessages.horizontalLine();
+        boolean isSuccessfullyMarked = targetTask.markAsDone();
+        if (isSuccessfullyMarked && toPrintMessage) {
+            DisplayMessages.printTaskMarked(targetTask);
+            DisplayMessages.horizontalLine();
+        }
+    }
+
+    public static void unmarkTask(int taskIndex) {
+        Task targetTask = userList.get(taskIndex);
+        DisplayMessages.horizontalLine();
+        boolean isSuccessfullyUnmarked = targetTask.markAsUndone();
+        if (isSuccessfullyUnmarked) {
+            DisplayMessages.printTaskUnmarked(targetTask);
+        }
+    }
+
+    public static void removeTask(int taskIndex) {
+        Task taskRemoved = userList.get(taskIndex);
+        userList.remove(taskIndex);
+        DisplayMessages.taskDeleted(taskRemoved, userList.size());
+    }
+
+    protected static void parseSavedData(String data) {
+        String[] dataArray = data.split(" ");
+        String taskType = dataArray[0];
+        int separatorIndex = Formatter.indexOf(dataArray, "|");
+        String description = "";
+        String date = "";
+        boolean taskIsDone = dataArray[1].equals("YES") ? true : false;
+        if (separatorIndex == -1) {
+            description = Formatter.parseUserInput(dataArray, 2, dataArray.length);
+        } else {
+            //saved Task is either an event or deadline
+            description = Formatter.parseUserInput(dataArray, 2, separatorIndex);
+            date = Formatter.parseUserInput(dataArray, separatorIndex+1, dataArray.length);
+        }
+
+        switch (taskType) {
+        case ("T"):
+            userList.add(new Task(description));
+            break;
+        case("D"):
+            userList.add(new Deadline(description, date));
+            break;
+        case("E"):
+            userList.add(new Event(description, date));
+            break;
+        default:
+            break;
+        }
+
+        if (taskIsDone) {
+            markTask(userList.size()-1, false);
+        }
+    }
+
+    public static void loadFile(File savedFile) {
+        try {
+            Scanner fileReader = new Scanner(savedFile);
+            while (fileReader.hasNextLine()) {
+                String data = fileReader.nextLine();
+                parseSavedData(data);
+            }
+            fileReader.close();
+        } catch (FileNotFoundException e) {
+            DisplayMessages.loadError();
+        }
+    }
+
+    protected static String getData() {
+        StringBuffer data = new StringBuffer();
+        for (Task t : userList) {
+            String temp = t.exportData() + "\n";
```

Comment:
Consider renaming the attribute `temp`, as it is not very clear what it represents solely from the name.